sdk/storage/Azure.Storage.Blobs/tests/BlobClientTests.cs

```diff
@@ -288,6 +288,99 @@ async Task Verify(Task<Response<BlobContentInfo>> upload)
             }
         }
 
+        [Test]
+        [TestCase(1)]
+        public async Task UploadAsync_File_AccessTier(int? maximumThreadCount)
+        {
+            using (this.GetNewContainer(out var container))
+            {
+                var blob = this.InstrumentClient(container.GetBlobClient(this.GetNewBlobName()));
+                var data = this.GetRandomBuffer(Constants.KB);
+
+                using (var stream = new MemoryStream(data))
+                {
+                    var path = Path.GetTempFileName();
+
+                    try
+                    {
+                        File.WriteAllBytes(path, data);
+
+                        var file = new FileInfo(path);
+
+                        var options = new ParallelTransferOptions { MaximumThreadCount = maximumThreadCount };
+
+                        await Verify(blob.UploadAsync(
```

Comment:
Nit - you probably don't need the `Verify` function if you're only doing one call.  This pattern is primarily for scenarios where we've got four or five values to test out.